src/sage/combinat/integer_vector.py

```diff
@@ -1163,7 +1198,18 @@ def rank(self, x):
             r += binomial(k + n - 1, k)
 
         return r
-
+            
+    def cardinality(self):
+        """
+        Return the cardinality of ``self``.
+        
+        TESTS::
+        
+            sage: IntegerVectors(200,5).cardinality()
```

Comment:
This is not really a test, but an example (demonstrating that cardinality works for large numbers).  It would also be good if there is an example that can be checked by hand, or even better, in your head.

A test would be `len(list(IntegerVectors(10, 5))) == IntegerVectors(10, 5).cardinality()`.

Warning: `len` automatically calls `cardinality`, if available.

Reply:
For an example&test that is checkable by hand (for us mortals) or in your head (for Gauss), you could have
```
sage: IntegerVectors(99, 3).cardinality()
5050
```
The "check in your head" would be that if the sum of the last two elements is $x$, then there are $x+1$ ways of dividing $x$ into `v[1]+v[2]`, namely, $0+x, 1+(x-1), \ldots, (x+0)$. Since $x$ can range from $0$ to $n=99$, the total cardinality is $1+2+\ldots+100 = 50 \times (1+100) = 5050$. Readers familiar with the [Gauss story](https://www.americanscientist.org/article/gausss-day-of-reckoning) will be amused.

You can then make a bigger demonstration with
```
sage: IntegerVectors(10^9 - 1, 3).cardinality()
500000000500000000
```